edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/ModuleClientProvider.cs

```diff
@@ -61,7 +61,7 @@ public class ModuleClientProvider : IModuleClientProvider
             ISdkModuleClientProvider sdkModuleClientProvider,
             Option<UpstreamProtocol> upstreamProtocol,
             Option<IWebProxy> proxy,
-            Option<string> productInfo,
+            string productInfo,
```

Comment:
Do you need to check not null for productInfo since type is changed from option<string> to string?

Reply:
+1 for null check, esp if moduleClient.SetProductInfo throws